ambari-server/src/main/java/org/apache/ambari/server/security/authentication/AmbariUserAuthentication.java

```diff
@@ -19,30 +19,31 @@
 
 import java.util.Collection;
 
-import org.apache.ambari.server.security.authorization.AmbariGrantedAuthority;
-import org.apache.ambari.server.security.authorization.User;
-import org.apache.ambari.server.security.authorization.UserIdAuthentication;
 import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
 
-public class AmbariUserAuthentication implements Authentication, UserIdAuthentication {
+public class AmbariUserAuthentication implements Authentication {
 
   private String serializedToken;
-  private User user;
-  private Collection<AmbariGrantedAuthority> userAuthorities;
-  private boolean authenticated = false;
+  private AmbariUserDetails userDetails;
+  private boolean authenticated;
```

Comment:
Should not these fields be final?

Reply:
Yup.. they can be.  The ones I replaced were not, but I will make them private since it is the right thing to do.